Library/Homebrew/requirements/java_requirement.rb

```diff
@@ -45,16 +30,69 @@ def inspect
 
   def display_s
     if @version
-      if @version[-1] == "+"
-        op = ">="
-        version = @version[0, @version.length-1]
-      else
+      if exact_version?
         op = "="
-        version = @version
+      else
+        op = ">="
       end
-      "#{name} #{op} #{version}"
+      "#{name} #{op} #{version_num}"
     else
       name
     end
   end
+
+  private
+
+  def version_num
+    if exact_version?
+      @version
+    else
+      @version[0, @version.length-1]
+    end
+  end
+
+  def exact_version?
+    @version && @version[-1] != "+"
+  end
+
+  def setup_java
+    java = preferred_java
+    return unless java
+    @java = java
+    @java_home = java.dirname.dirname
+  end
+
+  def preferred_java
+    if ENV["JAVA_HOME"]
+      Pathname.new(ENV["JAVA_HOME"])/"bin/java"
+    else
+      which "java"
+    end
+  end
+
+  def env_java
+    java_home = Pathname.new(@java_home)
+    ENV["JAVA_HOME"] = java_home
+    ENV.prepend_path "PATH", java_home/"bin"
+  end
+
+  def env_oracle(os)
```

Comment:
> It's not clear to me the difference between java and oracle.

`env_java` is environment setup common to any JDK (Oracle's, Apple's, OpenJDK, alternatives...) whereas `env_oracle` is particular to the Oracle JDK.

> Could do with a better name here.

Yeah, I agree. How about `env_java_common` and `env_oracle_jdk`? Wordier, but clearer too IMO.

> I'd also consider passing os as a symbol which will still interpolate correctly.

:+1: The calls to this function looked oddly out of place.